### ISSUE TITLE
Consider adding XMLAppendable.invoke

### ISSUE THREAD

**Issue by dmcg** [etiennestuder/gradle-jooq-plugin#702]
Just spitballing here, but converting a build.gradle to build.gradle.kts, having to add all the `apply`s is a bit annoying.

```kotlin
operator fun <T: XMLAppendable> T.invoke(block: T.() -> Unit) = this.apply(block)
```

lets me leave them out, but seems safe in other contexts?

**Comment by lukaseder** [comment]
I don't think you have to do anything extra. Just use the `jooq-meta-kotlin` library:

- https://github.com/etiennestuder/gradle-jooq-plugin/issues/62#issuecomment-1840475801

**Comment by dmcg** [comment]
> Do you know if there is a way to eliminate the apply() calls right after the `XMLAppendable` is instantiated?
> 
> ```
> properties = listOf(
>   Property().apply {
>     key = "PAGE_SIZE"
>     value = "2048"
>   }
> )
> ```

I don't really know why what you want doesn't work, but it doesn't.

You can write eg 
```
properties.add(
    Property().invoke {
        key = "ssl"
        value = "false"
    })
```

or 

```
properties.add(
    (Property()) {
        key = "ssl"
        value = "false"
    })
```

but I see that my build file has

```
properties.add(Property().withKey("ssl").withValue("false"))
```

**Comment by etiennestuder** [comment]
@dmcg @lukaseder Thanks for the feedback. I've integrated all of it into the README and examples. The DSL looks nice with these improvements, precisely like the Groovy-based DSL.

I tried to apply the `jooq-meta-kotlin` library to a Gradle build script containing a jOOQ configuration, but it led to some issues. I did not investigate it further, but I believe it's due to how my jOOQ plugin instantiates the top-level jOOQ Configuration object vs. specific extensions the jooq meta library declares.

I will leave things as they are for the time being. If my jOOQ plugin stops supporting the Groovy DSL, delegating the core jOOQ configuration to the `jooq-meta-kotlin` library should be aimed for.

**Comment by etiennestuder** [comment]
@dmcg @lukaseder The gradle-jooq-plugin v10.1 comes with first-class support for the Gradle Kotlin DSL.
https://github.com/etiennestuder/gradle-jooq-plugin/releases/tag/v10.1

It relies on Lukas's `jooq-meta-kotlin` library.
The only thing required in the user's build file is the declaration of `import org.jooq.meta.kotlin.*`
